core/dbt/tests/util.py

```diff
@@ -146,3 +147,204 @@ def run_sql_with_adapter(adapter, sql, fetch=None):
                     raise
                 finally:
                     conn.transaction_open = False
+
+
+class TestProcessingException(Exception):
+    pass
+
+
+def relation_from_name(adapter, name: str):
+    """reverse-engineer a relation (including quoting) from a given name and
+    the adapter. Assumes that relations are split by the '.' character.
+    """
+
+    # Different adapters have different Relation classes
+    cls = adapter.Relation
+    credentials = adapter.config.credentials
+    quote_policy = cls.get_default_quote_policy().to_dict()
```

Comment:
Well that makes sense, since this code was copied from dbt-adapter-tests.

Reply:
What would be a better way to do that?

Reply:
Not sure of a good general answer tbh. In that issue, it seems the user was able to get it working in their specific case by switching around a few lines in the method. The thing that gives me anxiety is recreating internal core logic within a method needed for end-to-end/integration tests.

The new framework still offers a big net improvement. The logic is _much_ less obfuscated, easier to find when debugging, and users can override/reimplement the `relation_from_name` utility if they needed to — right? or would they need to reimplement all utilities that depend on it, too?

Reply:
Oh, I see. Instead of using the quote_policy that's constructed there they went back to using the default_quote_policy. Tbh, I'm not exactly sure why we're developing that non-default quote policy there. Will have to investigate more, or at least document it.

Reply:
This has been fixed by removing the code that changes the default quote policy depending on the quotes around the identifier name.